tx/textual/valuerenderer/valuerenderer.go

```diff
@@ -46,58 +46,61 @@ func NewTextual(q CoinMetadataQueryFn) Textual {
 }
 
 // GetValueRenderer returns the value renderer for the given FieldDescriptor.
-func (r Textual) GetValueRenderer(fd protoreflect.FieldDescriptor) (ValueRenderer, error) {
+func (r Textual) GetValueRenderer(fd protoreflect.FieldDescriptor) (vr ValueRenderer, err error) {
+	md := fd.Message()
+
 	switch {
 	// Scalars, such as sdk.Int and sdk.Dec encoded as strings.
-	case fd.Kind() == protoreflect.StringKind && proto.GetExtension(fd.Options(), cosmos_proto.E_Scalar) != "":
-		{
+	case fd.Kind() == protoreflect.StringKind:
+		if proto.GetExtension(fd.Options(), cosmos_proto.E_Scalar) != "" {
 			scalar, ok := proto.GetExtension(fd.Options(), cosmos_proto.E_Scalar).(string)
 			if !ok || scalar == "" {
 				return nil, fmt.Errorf("got extension option %s of type %T", scalar, scalar)
 			}
 
-			vr := r.scalars[scalar]
+			vr = r.scalars[scalar]
 			if vr == nil {
 				return nil, fmt.Errorf("got empty value renderer for scalar %s", scalar)
 			}
-
-			return vr, nil
+		} else {
+			vr = NewStringValueRenderer()
 		}
+
 	case fd.Kind() == protoreflect.BytesKind:
-		return NewBytesValueRenderer(), nil
+		vr = NewBytesValueRenderer()
 
 	// Integers
 	case fd.Kind() == protoreflect.Uint32Kind ||
 		fd.Kind() == protoreflect.Uint64Kind ||
 		fd.Kind() == protoreflect.Int32Kind ||
 		fd.Kind() == protoreflect.Int64Kind:
-		{
-			return NewIntValueRenderer(), nil
-		}
-
-	case fd.Kind() == protoreflect.StringKind:
-		return stringValueRenderer{}, nil
+		vr = NewIntValueRenderer()
 
 	case fd.Kind() == protoreflect.MessageKind:
-		md := fd.Message()
 		fullName := md.FullName()
 
-		vr, found := r.messages[fullName]
+		rend, found := r.messages[fullName]
 		if found {
-			return vr, nil
+			vr = rend
+		} else {
+			vr = NewMessageValueRenderer(&r, md)
 		}
+
 		if fd.IsMap() {
 			return nil, fmt.Errorf("value renderers cannot format value of type map")
 		}
-		if fd.IsList() {
-			// This will be implemented in https://github.com/cosmos/cosmos-sdk/issues/12714
-			return nil, fmt.Errorf("repeated field renderer not yet implemented")
-		}
-		return NewMessageValueRenderer(&r, md), nil
 
 	default:
 		return nil, fmt.Errorf("value renderers cannot format value of type %s", fd.Kind())
 	}
+
+	// Coins are handled differently from other repeated values
+	if fd.IsList() && (md == nil || md.FullName() != (&basev1beta1.Coin{}).ProtoReflect().Descriptor().FullName()) {
+		vr = NewRepeatedValueRenderer(&r, md, string(fd.Name()), fd.Kind().String(), vr)
```

Comment:
Instead of passing `vr` here intoNewRepeatedValueRenderer,  a better idea is to simply pass `*Textual` as done in message value renderer.

This way, we can also keep `return`s :)

Reply:
resolving this since repeated has been refactored out